templates/federated_reporting/import.sh

```diff
@@ -43,36 +43,65 @@ dump_files="$(ls -1 "$CFE_FR_SUPERHUB_IMPORT_DIR/"*".sql.$CFE_FR_COMPRESSOR_EXT"
     exit 0
   }
 
-# make sure the script we are about to run is executable
-chmod u+x "$(dirname "$0")/import_file.sh"
+table_whitelist=$(printf "'%s'," $CFE_FR_TABLES | sed -e 's/,$//')
 
-log "Importing files: $dump_files"
 failed=0
-# for now, import in serial to avoid deadlocks (ENT-4742)
+log "Setting up schemas for import"
 for file in $dump_files; do
-  "$(dirname "$0")/import_file.sh" $file || failed=1
+  hostkey=$(basename "$file" | cut -d. -f1)
+  "$CFE_BIN_DIR"/psql -U $CFE_FR_DB_USER -d cfdb --set "ON_ERROR_STOP=1" \
+                      -c "SELECT ensure_feeder_schema('$hostkey', ARRAY[$table_whitelist]);" \
+    > schema_setup.log 2>&1 || failed=1
 done
+if [ "$failed" = "0" ]; then
+  log "Setting up schemas for import: DONE"
+else
+  log "Setting up schemas for import: FAILED"
+  # XXX: this needs to revert
```

Comment:
The ensure_feeder_schema() function acts as a transaction. Any changes that are made in that function will be rolled back if there is an error. So no need to revert.